sql/core/src/main/scala/org/apache/spark/sql/execution/stat/StatFunctions.scala

```diff
@@ -288,7 +288,7 @@ object StatFunctions extends Logging {
     }
 
     // If there is no selected columns, we don't need to run this aggregate, so make it a lazy val.
-    lazy val aggResult = ds.select(aggExprs: _*).queryExecution.toRdd.collect().head
+    lazy val aggResult = ds.select(aggExprs: _*).queryExecution.toRdd.map(_.copy()).collect().head
```

Comment:
Hm, we're taking the head row anyway. Do you have any e2e example that produces a wrong result?

Reply:
+1, I think we'd better add a test for it

Reply:
We encountered this issue when passing gluten [ut](https://github.com/apache/spark/blob/0515e6b96fde80f72c5bcedf0d02884bd46450ab/sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala#L2609). Because in Gluten we rewritten [ColumnarToRow](https://github.com/oap-project/gluten/blob/d9c38e6897f0b94687934cbf5ac68c12cda2bc96/gluten-data/src/main/scala/org/apache/spark/sql/execution/GlutenColumnarToRowExec.scala#L188). which will release the row after the row is used. It seems that this issue is hard to reproduce on apache spark. Because the ColumnarToRow of apache spark is in on heap memory, which is recycled by GC. Do you have any suggestion to reproduce this issue in apache spark? @HyukjinKwon @cloud-fan @zhengruifeng

Reply:
This is only a bug if Spark installs third-party physical operators that release memory eagerly.

Reply:
According to the doc of `QueryExecution.toRDD`, I think adding copy is the right thing to do
```
  /**
   * Internal version of the RDD. Avoids copies and has no schema.
   * Note for callers: Spark may apply various optimization including reusing object: this means
   * the row is valid only for the iteration it is retrieved. You should avoid storing row and
   * accessing after iteration. (Calling `collect()` is one of known bad usage.)
   * If you want to store these rows into collection, please apply some converter or copy row
   * which produces new object per iteration.
   * Given QueryExecution is not a public class, end users are discouraged to use this: please
   * use `Dataset.rdd` instead where conversion will be applied.
   */
  lazy val toRdd: RDD[InternalRow] = new SQLExecutionRDD(
    executedPlan.execute(), sparkSession.sessionState.conf)
```
It happens to work here because the result only has one row (it's a global aggregate). I'm fine without testing it as this follows the guidance of `QueryExecution.toRdd` doc

Reply:
Thanks @cloud-fan  for the valuable information.  @HyukjinKwon @zhengruifeng can we follow the suggestion from wenchen to merge this PR without testing?

Reply:
SGTM

Reply:
@HyukjinKwon Do you have any further comment? Thanks.